dropshot/src/handler.rs

```diff
@@ -1436,6 +1436,172 @@ impl From<HttpResponseUpdatedNoContent> for HttpHandlerResult {
     }
 }
 
+/** Describes headers associated with a 300-level response. */
+#[derive(JsonSchema, Serialize)]
+#[doc(hidden)]
+pub struct RedirectHeaders {
+    /** HTTP "Location" header */
+    /*
+     * What type should we use to represent header values?
+     *
+     * It's tempting to use `http::HeaderValue` here.  But in HTTP, header
+     * values can contain bytes that aren't valid Rust strings.  See
+     * `http::header::HeaderValue`.  We could propagate this nonsense all the
+     * way to the OpenAPI spec, encoding the Location header as, say,
+     * base64-encoded bytes.  This sounds really annoying to consumers.  It's
+     * also a fair bit more work to implement.  We'd need to create a separate
+     * type for this field so that we can impl `Serialize` and `JsonSchema` on
+     * it, and we'd need to also impl serialization of byte sequences in
+     * `MapSerializer`.  Ugh.
+     *
+     * We just use `String`.  This might contain values that aren't valid in
+     * HTTP response headers.  But we can at least validate that at runtime, and
+     * it sure is easier to implement!
+     */
+    location: String,
+}
+
+/** See `http_response_found()` */
+pub type HttpResponseFound =
+    HttpResponseHeaders<HttpResponseFoundStatus, RedirectHeaders>;
+
+/**
+ * `http_response_found` returns an HTTP 302 "Found" response with no response
+ * body.
+ *
+ * The sole argument will become the value of the `Location` header.  This is
+ * where you want to redirect the client to.
+ *
+ * Per MDN and RFC 9110 S15.4.3, you might want to use 307 ("Temporary
+ * Redirect") or 303 ("See Other") instead.
+ */
+pub fn http_response_found(
```

Comment:
why `http_response_found` rather than `http_found` to be consistent with the others? because `http_found` would be confusing? Fair enough.